File: iosApp/iosApp/designsystem/data/Spacings.swift
import Foundation
import shared

public class Spacings {
    let xTiny: CGFloat
    let tiny: CGFloat
    let small: CGFloat
    let medium: CGFloat
    let large: CGFloat
    let xLarge: CGFloat
    let xxLarge: CGFloat
    let xxxLarge: CGFloat

    init(_ sharedSpacing: shared.Spacings) {
        xTiny = sharedSpacing.xTiny.cgFloat
        tiny = sharedSpacing.tiny.cgFloat
        small = sharedSpacing.small.cgFloat
        medium = sharedSpacing.medium.cgFloat
        large = sharedSpacing.large.cgFloat
        xLarge = sharedSpacing.xLarge.cgFloat
        xxLarge = sharedSpacing.xxLarge.cgFloat
        xxxLarge = sharedSpacing.xxxLarge.cgFloat
    }

    init(
        xTiny: Int,
        tiny: Int,
        small: Int,
        medium: Int,
        large: Int,
        xLarge: Int,
        xxLarge: Int,
        xxxLarge: Int
    ) {
        self.xTiny = CGFloat(xTiny)
        self.tiny = CGFloat(tiny)
        self.small = CGFloat(small)
        self.medium = CGFloat(medium)
        self.large = CGFloat(large)
        self.xLarge = CGFloat(xLarge)
        self.xxLarge = CGFloat(xxLarge)
        self.xxxLarge = CGFloat(xxxLarge)
    }
}

File: buildSrc/src/main/kotlin/de/stefan/lang/designsystem/spacing/Spacings.kt
package de.stefan.lang.designsystem.spacing

import de.stefan.lang.designsystem.animation.AnimationDurations
import de.stefan.lang.designsystem.core.PropertyReader
import de.stefan.lang.designsystem.shapes.Shape
import org.gradle.kotlin.dsl.provideDelegate

data class Spacings(
    val xTiny: Int,
    val tiny: Int,
    val small: Int,
    val medium: Int,
    val large: Int,
    val xLarge: Int,
    val xxLarge: Int,
    val xxxLarge: Int,
) {
    val all: Map<String, Int> by lazy {
        PropertyReader.read<Int, Spacings>(this)
    }

    val allSorted by lazy {
        all
            .entries
            .sortedWith (
                compareBy{ it.value }
            )
            .associate { it.key to it.value }
    }
}
